source/Xunit.Gherkin.Quick/CoreModel/Feature/Discovery/FeatureDiscoveryModel.cs

```diff
@@ -11,18 +13,31 @@ public FeatureDiscoveryModel(IFeatureFileRepository featureFileRepository)
             _featureFileRepository = featureFileRepository ?? throw new ArgumentNullException(nameof(featureFileRepository));
         }
 
-        public global::Gherkin.Ast.Feature Discover(Type featureClassType)
+        public IEnumerable<Tuple<string,global::Gherkin.Ast.Feature>> Discover(Type featureClassType)

```

Comment:
I've choosen IEnumerable since many Xunit Library such as the Discoverer which invokes it, uses IEnumerable. 

Since I am not using anything from the List semantic, I return the less restrictive type. Also it prevents anyone from adding elements.

Reply:
I understand your point, but it violates robustness principle: it's actually better to return concrete type and take general type from the code design perspective. Xunit is not an exemplary design reference in this case, not the best one to follow, with all due respect.
Also, when returning IEnumerable, the list is not materialized inside the method, but instead where the enumeration is necessary. If the materialization causes the exception to be thrown, the stack trace will point at the materialization line in code, not this method, so it's a maintenance headache as well.
Adding the item to the list is not a concern of the method, the caller needs to take care of the returned value, so that is also not a problem.
With all these points, please change it to List.